cclib/method/bader.py

```diff
@@ -74,21 +74,28 @@ def calculate(self, indices=None, fupdate=0.05):
            Cartesian, uniformly spaced grids are assumed for this function.
            """
 
-        # First obtain charge densities on the grid
-        if len(self.data.mocoeffs) == 1:
-            self.chgdensity = electrondensity_spin(
-                self.data, self.volume, [self.data.mocoeffs[0][: self.data.homos[0]]]
-            )
-            self.chgdensity.data *= 2
+        # Obtain charge densities on the grid if it does not contain one.
+        if self.volume.data.sum() == 0.0:
```

Comment:
Do you think there's a better way to check this?

Reply:
Yes -- `numpy.any` should do the job. Thanks for pointing this out!